Sources/NIOSSL/SSLContext.swift

```diff
@@ -211,6 +203,72 @@ private func clientPSKCallback(ssl: OpaquePointer?,
     return UInt32(clientPSK.count)
 }
 
+private func sslContextCallback(ssl: OpaquePointer?, arg: UnsafeMutableRawPointer?) -> Int32 {
+    // This should be a safe force unwrap. If ssl is not set its a singal of a larger problem
+    let parentSwiftContext = NIOSSLContext.lookupFromRawContext(ssl: ssl!)
+
+    // Check if we have a callback result set from the SSLHandler
+    if let sslContextCallbackResult = parentSwiftContext.sslContextCallbackResult {
+        let userChosenContext: NIOSSLContext
+        switch sslContextCallbackResult {
+        case .success(let value):
+            userChosenContext = value
+        case .failure:
+            return SSL_TLSEXT_ERR_NOACK
+        }
+
+        // The user may or may not retain a reference to the NIOSSLContext
```

Comment:
Is this necessary now that we are using an EventLoopFuture? We are holding on to the future and the future I think holds a strong reference to the context right?

Reply:
That's true, but reveals a concern: we can end up with a reference cycle that needs to be explicitly broken. So I think we want to keep the code we have here, and drop the stored future. It's not that useful to us anyway, because we only want the ability to attach a callback to it once, allowing us to safely drop it once we've done that.

This also takes us closer to the `CustomVerifyManager` pattern.